Framework/Sources/FloatingPanel.swift

```diff
@@ -424,43 +444,53 @@ class FloatingPanel: NSObject, UIGestureRecognizerDelegate {
             }
         }
 
-        guard
-            state == layoutAdapter.topMostState,   // When not top most(i.e. .full), don't scroll.
-            interactionInProgress == false,        // When interaction already in progress, don't scroll.
-            surfaceView.frame.minY == layoutAdapter.topY
-        else {
+        guard self.isFloatingPanelAtTopMostPosition() && self.interactionInProgress else {
             return false
         }
 
         // When the current and initial point within grabber area, do scroll.
-        if grabberAreaFrame.contains(point), !grabberAreaFrame.contains(initialLocation) {
+        if grabberAreaFrame.contains(point) && !grabberAreaFrame.contains(initialLocation) {
             return true
         }
 
-        guard
-            scrollView.frame.contains(initialLocation), // When initialLocation not in scrollView, don't scroll.
-            !grabberAreaFrame.contains(point)           // When point within grabber area, don't scroll.
-        else {
+        // When initialLocation not in scrollView and when point within grabber area, don't scroll.
+        if !scrollView.frame.contains(initialLocation) && grabberAreaFrame.contains(point) {
```

Comment:
It looks like you missed the most important part of [my original comment](https://github.com/pickware/FloatingPanel/pull/1#discussion_r322675858)
```suggestion
        if !scrollView.frame.contains(initialLocation) || grabberAreaFrame.contains(point) {
```

Reply:
Yep...sorry. Had a reasoning error there.